### PR TITLE
Enable transfer back no payable check

### PR DESCRIPTION
**Pull request by sasurobert**
Enable transfer back no payable check

### CODE REVIEW

**Review by andrei-marinica — APPROVED** [review]

File: vmhost/hostCore/execution.go [hunk 1]
```diff
@@ -348,6 +349,47 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
...
+
+		esdtTransfers, err := host.esdtTransferParser.ParseESDTTransfers(child, parent, function, args)
+		if err != nil {
+			continue
+		}
+
+		if esdtTransfers.CallFunction != "" {
+			continue
+		}
+
```
Comment (camilbancioiu) [inline vmhost/hostCore/execution.go:388]: Optionally, consider extracting as a separate method, returning only a Boolean to add the transfer or not.

**Review by camilbancioiu — APPROVED** [review]

**Review by matei-p — APPROVED** [review]

**Review by camilbancioiu — COMMENTED** [review]
TODO unit & integration tests

File: vmhost/hostCore/execution.go (hunk 1 above)
Comment (sasurobert) [reply vmhost/hostCore/execution.go]: done

**Review by andrei-marinica — DISMISSED** [review]

File: vmhost/contexts/async_test.go [hunk 2]
```diff
@@ -835,7 +835,7 @@ func createCallbackInput(t *testing.T, lastTransfer *vmcommon.OutputTransfer) (*
 	return createCallbackInputWithVMOutput(t, vmOutput, lastTransfer)
 }
 
-func createCallbackInputWithVMOutput(t *testing.T, vmOutput *vmcommon.VMOutput, lastTransfer *vmcommon.OutputTransfer) (*vmcommon.ContractCallInput, *vmcommon.ContractCallInput, uint64) {
+func createCallbackInputWithVMOutput(t *testing.T, vmOutput *vmcommon.VMOutput, _ *vmcommon.OutputTransfer) (*vmcommon.ContractCallInput, *vmcommon.ContractCallInput, uint64) {
```
Comment (matei-p) [inline vmhost/contexts/async_test.go:838]: what is the point of an ignored function argument?

**Review by matei-p — CHANGES_REQUESTED** [review]

File: vmhost/contexts/async_test.go (hunk 2 above)
Comment (sasurobert) [reply vmhost/contexts/async_test.go:820]: fixed the linter issue.

**Review by matei-p — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**